Realm/RLMArrayLinkView.mm

```diff
@@ -46,19 +58,32 @@ @implementation RLMArrayLinkView {
     std::unique_ptr<RLMObservationInfo> _observationInfo;
 }
 
-- (RLMArrayLinkView *)initWithParent:(__unsafe_unretained RLMObjectBase *const)parentObject
-                            property:(__unsafe_unretained RLMProperty *const)property {
+- (RLMArrayLinkView *)initWithList:(realm::List)list
+                             realm:(__unsafe_unretained RLMRealm *const)realm
+                        parentInfo:(RLMClassInfo *)parentInfo
+                          property:(__unsafe_unretained RLMProperty *const)property {
     self = [self initWithObjectClassName:property.objectClassName];
     if (self) {
-        _realm = parentObject->_realm;
-        _backingList = realm::List(_realm->_realm, parentObject->_row.get_linklist(parentObject->_info->tableColumn(property)));
-        _objectInfo = &parentObject->_info->linkTargetType(property.index);
-        _ownerInfo = parentObject->_info;
+        _realm = realm;
+        REALM_ASSERT_DEBUG(list.get_realm() == realm->_realm);
```

Comment:
Doesn't need to be a `_DEBUG()` assert (the intention for _DEBUG is for things we would like to always check, but are too slow to do in real builds).